src/main/java/org/osc/manager/ism/api/util/ValidationUtil.java

```diff
@@ -42,4 +43,12 @@ public DeviceEntity getDeviceOrThrow(String deviceId) {
         }
         return device;
     }
+
+    public void validateParentIdMatches(DeviceEntity entity, Long parentId)
+            throws Exception {
+        if (!parentId.equals(Long.parseLong(entity.getId()))) {
+            throw new IllegalArgumentException(
```

Comment:
please add something like String.format("The ID %d specified in the '%s' data does not match the id specified in the URL", id, objName)

> org.osc.core.broker.rest.server.ApiUtil.createIdMismatchException(Long, String)